force-app/main/default/classes/ContactController.cls

```diff
@@ -1,170 +1,140 @@
 public with sharing class ContactController {
 
-    private static Boolean check = true;
-    private static List<String> closedStatus = new List<String>{'Closed'};
-    private static List<String> unclosedStatus = new List<String>{'New', 'Working', 'Escalated'};
-    private static List<String> allStatuses = new List<String>{'New', 'Working', 'Escalated', 'Closed'};
+    public class NewExecutorNotFoundException extends Exception{}
+    private static Integer maxCases = 0;
+    private static final Integer MAX_CONTACTS = 50000;
+    private static final String ERROR_MESSAGE = 'first error: DELETE_FAILED, Your attempt to delete contact with Id {0} could not be completed because it is associated with some cases';
+    @testVisible
+    private static String errorMessageTest = '';
 
-
-    @AuraEnabled
-    public static List<Contact> start(List<Id> source){
-        if(check){
-            System.debug('Inside start method');
-            return deleteContacts(source);
-        }else{
-            return bung(source);
-        }
+    @AuraEnabled(cacheable = true)
+    public static List<Contact> getAllContacts(){
+        return [SELECT Id, FirstName, LastName, a.Name, Phone, Email
+        FROM Contact c, c.Account a            
+        WITH SECURITY_ENFORCED ORDER BY LastName LIMIT :MAX_CONTACTS];
     }
 
-    @AuraEnabled
-    public static List<Contact> bung(Id[] lstConIds){
-        try {
-            list<Contact> lstConsToDelete = new list<Contact>();
-            for(Id idCon : lstConIds) {
-                lstConsToDelete.add(new Contact(Id = idCon));
-            }
-            if(!lstConsToDelete.isEmpty()) {
-                delete lstConsToDelete;
+    @AuraEnabled(cacheable = false)
+    public static List<Contact> start(List<Contact> selectedContacts){
+        if (selectedContacts == null) {
+            return null;
+        }else {
+            try {
+                List<Case> updateCases = new List<Case>();
```

Comment:
Variables should be declared where they firstly used.

Reply:
Fixed, in my opinion